openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -536,6 +536,13 @@ def post(self, request):
         redirect_url = get_redirect_url_with_host(root_url, redirect_to)
         response = self._create_response(request, {}, status_code=200, redirect_url=redirect_url)
         set_logged_in_cookies(request, response, user)
+        if not user.is_active:
+            response.set_cookie(
+                settings.SHOW_ACTIVATE_CTA_POPUP_COOKIE_NAME,
+                True,
+                domain=settings.SESSION_COOKIE_DOMAIN,
+                secure=request.is_secure()
+            )  # setting the cookie for account activation dialogue
```

Comment:
Add a little bit more details why we are using a cookie, like; since this dialogue will appear on some MFEs as well.